src/components/PostsList.tsx

```diff
@@ -1,86 +1,53 @@
 import React from 'react';
-
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
-
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
-
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
-
-          <td data-cy="PostTitle">
-            fugit voluptas sed molestias voluptatem provident
-          </td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
-
-        <tr data-cy="Post">
-          <td data-cy="PostId">18</td>
-
-          <td data-cy="PostTitle">
-            voluptate et itaque vero tempora molestiae
-          </td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link"
-            >
-              Close
-            </button>
-          </td>
-        </tr>
-
-        <tr data-cy="Post">
-          <td data-cy="PostId">19</td>
-          <td data-cy="PostTitle">adipisci placeat illum aut reiciendis qui</td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
-
-        <tr data-cy="Post">
-          <td data-cy="PostId">20</td>
-          <td data-cy="PostTitle">doloribus ad provident suscipit at</td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </div>
-);
+import { Post } from '../types/Post';
+import { PostItem } from './PostItem';
+
+type Props = {
+  posts: Post[];
+  currentPost: Post | null;
+  setCurrentPost: (p: Post | null) => void;
+};
+
+export const PostsList: React.FC<Props> = ({
+  posts,
+  currentPost,
+  setCurrentPost,
+}) => {
+  const selectPost = (post: Post) => {
+    if (currentPost?.id === post.id) {
+      setCurrentPost(null);
+
+      return;
+    }
+
+    setCurrentPost(post);
+  };
+
+  return (
+    <div data-cy="PostsList">
```

Comment:
If you have class PostList and the same component name why you don't use the list tag with `<li>`?

Reply:
because I use the starting markup that I receive with the assignment. it is already built in such a way as to obtain a design similar to the one in the sample that I need to repeat

Reply:
and the layout is exactly like this, because the Balma framework is used